File: backend/src/main/java/info/gamewise/lor/videos/port/in/GetVideosUseCase.java
package info.gamewise.lor.videos.port.in;

import info.gamewise.lor.videos.domain.LoRVideo;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface GetVideosUseCase {
    Page<LoRVideo> fetchByFilter(SearchParams params, Pageable pageable);

    record SearchParams(List<String> regions, List<String> champions,
                        List<String> channels) {
    }
}

File: backend/src/main/kotlin/info/gamewise/lor/videos/port/in/GetVideosUseCase.kt
package info.gamewise.lor.videos.port.`in`

import info.gamewise.lor.videos.domain.LoRVideo
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface GetVideosUseCase {
    fun fetchByFilter(params: SearchParams, pageable: Pageable): Page<LoRVideo>
}

data class SearchParams(val regions: List<String> = emptyList(),
                        val champions: List<String> = emptyList(),
                        val channels: List<String> = emptyList())